spikeinterface/sortingcomponents/motion_correction.py

```diff
@@ -188,63 +190,61 @@ def _get_closest_ind(array, values):
     temporal_bins: np.array
         Temporal bins in second.
     spatial_bins: None or np.array
         Bins for non-rigid motion. If None, rigid motion is used
-    direction: int in (0, 1, 2)
-        Dimension of shift in channel_locations.
-    spatial_interpolation_method : 'kriging' or 'idw' or 'nearest'
-        see `spikeinterface.preprocessing.get_spatial_interpolation_kernel()` for more detail
-        Choice of the method
+    direction: int (0, 1, 2)
+        Dimension along which channel_locations are shifted (0 - x, 1 - y, 2 - z), by default 1
+    spatial_interpolation_method: str
+        'kriging' or 'idw' or 'nearest'.
+        See `spikeinterface.preprocessing.get_spatial_interpolation_kernel()` for more details.
+        Choice of the method:
             'kriging' : the same one used in kilosort
-            'idw' : inverse  distance weithed
+            'idw' : inverse  distance weighted
             'nearest' : use nereast channel
-    sigma_um : float (default 20.)
+    sigma_um: float (default 20.)
         Used in the 'kriging' formula
     p: int (default 1)
         Used in the 'kriging' formula
     num_closest: int (default 3)
-        Used for 'idw'
+        Number of closest channels used by 'idw' method for interpolation.
     border_mode: 'remove_channels', 'force_extrapolate', 'force_zeros'
         Control how channels are handled on border
         * 'remove_channels': remove channels on the border, the recording has less channels
         * 'force_extrapolate': keep all channel and force extrapolation (can lead to strange signal)
         * 'force_zeros': keep all channel but set zeros when outside (force_extrapolate=False)
 
-
     Returns
     -------
-    Corrected_recording: CorrectMotionRecording
+    corrected_recording: CorrectMotionRecording
         Recording after motion correction
     """
     name = 'correct_motion'
 
     def __init__(self, recording, motion, temporal_bins, spatial_bins, direction=1,
-                border_mode='remove_channels',
-                spatial_interpolation_method='kriging', sigma_um=20., p=1, num_closest=3,
-
-                ):
-        assert recording.get_num_segments() == 1, 'correct is handle only for one segment for the moment'
+                 border_mode='remove_channels', spatial_interpolation_method='kriging',
+                 sigma_um=20., p=1, num_closest=3):
+        assert recording.get_num_segments() == 1, 'correct_motion() is only availabl for single-segment recordings'
         
         channel_locations = recording.get_channel_locations()
 
         spatial_interpolation_kwargs = dict(sigma_um=sigma_um, p=p, num_closest=num_closest)
         if border_mode == 'remove_channels':
+            assert channel_locations.ndim >= direction, (f"'direction' {direction} not available. "
+                                                         "Channel locations have {channel_locations.ndim} dimensions.")
```

Comment:
@alejoe91 this can be outside the if no ?

Reply:
yep, one sec!